src/python/pants/backend/docker/rules_test.py

```diff
@@ -0,0 +1,213 @@
+            MockGet(
+                output_type=DockerBuildContext,
+                input_type=DockerBuildContextRequest,
+                mock=build_context_mock,
+            ),
+            MockGet(
+                output_type=Snapshot,
+                input_type=Digest,
+                mock=lambda _: EMPTY_SNAPSHOT,
+            ),
+            MockGet(
+                output_type=ProcessResult,
+                input_type=Process,
+                # Process() generation has its own tests in test_docker_binary_build_image
+                mock=lambda _: ProcessResult(
+                    stdout=b"stdout",
+                    stdout_digest=EMPTY_FILE_DIGEST,
+                    stderr=b"stderr",
+                    stderr_digest=EMPTY_FILE_DIGEST,
+                    output_digest=EMPTY_DIGEST,
+                ),
+            ),
+        ],
+    )
+
+    assert result.digest == EMPTY_DIGEST
+    assert len(result.artifacts) == 1
+    assert result.artifacts[0].relpath is None
+
+    version = expected_features.get("version", "latest")
+    assert f"Built docker image: image:{version}" in result.artifacts[0].extra_log_lines
+
+
+def test_create_docker_build_context():
+    tgt = DockerImage(address=Address("src/test", target_name="image"), unhydrated_values={})
+    request = DockerBuildContextRequest(
+        address=Address("src/test", target_name="image"), context_root=".", targets=Targets([tgt])
+    )
+
+    result = run_rule_with_mocks(
+        create_docker_build_context,
+        rule_args=[request],
+        mock_gets=[
+            MockGet(
+                output_type=SourceFiles,
+                input_type=SourceFilesRequest,
+                mock=lambda _: SourceFiles(
+                    snapshot=EMPTY_SNAPSHOT,
+                    unrooted_files=tuple(),
+                ),
+            ),
+            MockGet(
+                output_type=Targets,
+                input_type=DependenciesRequest,
+                mock=lambda _: Targets([]),
+            ),
+            MockGet(
+                output_type=FieldSetsPerTarget,
+                input_type=FieldSetsPerTargetRequest,
+                mock=lambda request: FieldSetsPerTarget([[DockerFieldSet.create(tgt)]]),
+            ),
+            MockGet(
+                output_type=BuiltPackage,
+                input_type=PackageFieldSet,
+                mock=lambda _: BuiltPackage(EMPTY_DIGEST, []),
+            ),
+            MockGet(
+                output_type=Digest,
+                input_type=AddPrefix,
+                mock=lambda _: EMPTY_DIGEST,
+            ),
+            MockGet(
+                output_type=Digest,
+                input_type=MergeDigests,
+                mock=lambda _: EMPTY_DIGEST,
+            ),
+        ],
+        # need AddPrefix here, since UnionMembership.is_member() throws when called with non
+        # registered types
+        union_membership=UnionMembership({PackageFieldSet: [DockerFieldSet], AddPrefix: []}),
```

Comment:
This made me wonder whether it is correct to throw on a `is_member()` check.. it behaves more like an assert in that way..